sequencer/src/state.rs

```diff
@@ -526,9 +524,16 @@ async fn update_state_storage(
             .resolve()
             .context("failed to resolve to chain config")?;
 
-        if let Err(err) = storage.insert_chain_config(cf).await {
-            storage.revert().await;
-            return Err(err);
+        match storage.insert_chain_config(cf).await {
+            Err(err) => {
+                return Err(err);
+            }
+            Ok(()) => {
+                if let Err(err) = storage.commit().await {
+                    storage.revert().await;
+                    return Err(err.into());
+                }
+            }
```

Comment:
I just realized we have the `transaction` helper function in `persistence/sql.rs` which does all this commit/revert/error handling stuff for us. Maybe we should export that, use it in `insert_chain_config`, and remove this error handling here? WDYT?

Also, there appears to be a bug in that helper function where we don't `revert` if the `commit` fails (my bad)

Reply:
yeah that one takes Persistence.

Reply:
Oh yeah. I think it only uses `persistence.db` so hopefully it's generalizable

Reply:
Nice, this feels way more comfortable to me now 👍